syntax = "proto3";

package payment;

service PaymentService {
  rpc Prepare(PrepareRequest) returns (PrepareResponse);
  rpc Commit(CommitRequest) returns (CommitResponse);
  rpc Abort(AbortRequest) returns (AbortResponse);
}

message PrepareRequest {
  string order_id = 1;
  string order_details_json = 2;
  string message = 3;
}

message PrepareResponse {
  bool willCommit = 1;
  string message = 2;
}

message CommitRequest {
  string order_id = 1;
  string message = 2;
}

message CommitResponse {
  bool success = 1;
  string message = 2;
  string order_details_json = 3;
}

message AbortRequest {
  string order_id = 1;
  string message = 2;
}

message AbortResponse {
  bool success = 1;
  string message = 2;
  string order_details_json = 3;
}